addon/visit.js

```diff
@@ -51,13 +51,16 @@ class PageVisitor {
         this.timeout = options.timeout;
         this.otherOptions = options.otherOptions;
 
-        const trainee = await browser.runtime.sendMessage(
-            'fathomtrainees@mozilla.com',
-            {type: 'trainee',
-                traineeId: this.otherOptions.traineeId});
-
-        this.viewportHeight = trainee.viewportSize.height;
-        this.viewportWidth = trainee.viewportSize.width;
+        // If the subclass's form does not provide the viewport size, it should
+        // have a method to provide that information in a different way.
+        if (options.hasOwnProperty('viewportHeight')) {
```

Comment:
I was thinking of having the Corpus Collector implement getViewportHeightAndWidth() also, saving this branch. Thoughts?

Reply:
I think that's reasonable, and will go that route.